gcp/src/main/java/org/apache/iceberg/gcp/auth/GoogleAuthSession.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.gcp.auth;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.HTTPHeaders;
+import org.apache.iceberg.rest.HTTPRequest;
+import org.apache.iceberg.rest.ImmutableHTTPRequest;
+import org.apache.iceberg.rest.auth.AuthSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An authentication session that uses Google Credentials (typically Application Default
+ * Credentials) to obtain an OAuth2 access token and add it to HTTP requests.
+ */
+public class GoogleAuthSession implements AuthSession {
+  private static final Logger LOG = LoggerFactory.getLogger(GoogleAuthSession.class);
+  private final GoogleCredentials credentials;
+
+  /**
+   * Constructs a GoogleAuthSession with the provided GoogleCredentials.
+   *
+   * @param credentials The GoogleCredentials to use for authentication.
+   */
+  public GoogleAuthSession(GoogleCredentials credentials) {
+    Preconditions.checkArgument(credentials != null, "Invalid credentials: null");
+    this.credentials = credentials;
+  }
+
+  /**
+   * Authenticates the given HTTP request by adding an "Authorization: Bearer token" header. The
+   * access token is obtained from the GoogleCredentials.
+   *
+   * @param request The HTTPRequest to authenticate.
+   * @return A new HTTPRequest with the added Authorization header.
+   * @throws UncheckedIOException if an IOException occurs while refreshing the access token.
+   */
+  @Override
+  public HTTPRequest authenticate(HTTPRequest request) {
+    try {
+      // Ensure the credentials have a valid token, refreshing if necessary.
+      // refreshIfExpired() returns true if the token was refreshed, false otherwise.
+      // In either case, getAccessToken() after this call should provide a usable token.
```

Comment:
I don't think we need these contextual comments.  The behavior is clear and this isn't adding anything.